File: backend/app/schemas/attendance.py
from pydantic import BaseModel
from typing import Optional
from datetime import date


# POST /api/v1/attendance/check-in 응답
class CheckInResponse(BaseModel):
    checked_at: date
    current_streak: int
    max_streak: int
    message: str          # "출석 완료" / "이미 출석했습니다"


# GET /api/v1/attendance/status 응답
class AttendanceStatusResponse(BaseModel):
    today_checked: bool
    current_streak: int
    max_streak: int
    last_checked_at: Optional[date] = None


# GET /api/v1/attendance/calendar?year=2026&month=6 응답
class AttendanceCalendarResponse(BaseModel):
    year: int
    month: int
    checked_dates: list[date]   # 해당 월의 출석한 날짜 목록
    total_count: int            # 해당 월 총 출석일수

File: frontend/src/api/attendance.js
// 출석체크 API 모듈 — /api/v1/attendance
import { getAccessToken } from '../utils/token.js'

const base = () => import.meta.env.VITE_API_BASE_URL ?? '/api/v1'

function authHeaders() {
  const t = getAccessToken()
  return t ? { Authorization: `Bearer ${t}` } : {}
}

// POST /attendance/check-in → { checked_at, current_streak, max_streak, message }
export async function checkInAttendance() {
  const res = await fetch(`${base()}/attendance/check-in`, {
    method: 'POST',
    headers: { ...authHeaders() },
  })
  if (!res.ok) throw new Error(await res.text())
  return res.json()
}

// GET /attendance/status → { today_checked, current_streak, max_streak, last_checked_at }
export async function getAttendanceStatus() {
  const res = await fetch(`${base()}/attendance/status`, {
    headers: { ...authHeaders() },
  })
  if (!res.ok) throw new Error(await res.text())
  return res.json()
}

// GET /attendance/calendar?year&month → { year, month, checked_dates: [date], total_count }
export async function getAttendanceCalendar(year, month) {
  const res = await fetch(`${base()}/attendance/calendar?year=${year}&month=${month}`, {
    headers: { ...authHeaders() },
  })
  if (!res.ok) throw new Error(await res.text())
  return res.json()
}
